maze.pl~

```diff
@@ -0,0 +1,259 @@
+% 1 ╚╗╚╗╚╣
+% 2 ╞╩═╝╞╩
+%
+% The entrance the maze is in cell(0,0) at the upper left and the exit is at
+% cell(5,2).  The structure of this maze is codified using the follwoing rules:
+
+corridor(cell(2,0), cell(2,1)).
+corridor(cell(1,1), cell(1,2)).
+corridor(cell(4,1), cell(4,0)).
+corridor(cell(5,2), cell(5,1)).
+corridor(cell(1,1), cell(0,1)).
+corridor(cell(2,2), cell(3,2)).
+corridor(cell(0,2), cell(1,2)).
+corridor(cell(3,1), cell(2,1)).
+corridor(cell(3,0), cell(2,0)).
+corridor(cell(4,2), cell(5,2)).
+corridor(cell(4,1), cell(5,1)).
+corridor(cell(1,0), cell(2,0)).
+corridor(cell(5,1), cell(5,0)).
+corridor(cell(0,0), cell(0,1)).
+corridor(cell(3,1), cell(3,2)).
+corridor(cell(3,0), cell(4,0)).
+corridor(cell(2,2), cell(1,2)).
+
+% In these rules, each rule corridor(X,Y) states there is a corridor leading
+% from room X to room Y.
+
+
+% Step 1: Are two rooms connected?
+%
+% The corridor/2 rule only works one way.  Create a new rule connected/2 that
+% will be true if there is a corridor between the two rooms (in any direction).
+%
+
+connected(X,Y) :- corridor(X,Y);
+	          corridor(Y,X).
+
+
+
+:- begin_tests(connected).
+
+% Tests that 2.0 connects to 2,1 and vice-versa
+test(connected_forward) :- connected(cell(2,0), cell(2,1)), !.
+test(connected_backward) :- connected(cell(2,1), cell(2,0)), !.
+
+% Find all cells connected to/from 0,0
+test(connected_from_0_0, all(X == [cell(0,1)])) :- connected(cell(0,0), X).
+test(connected_to_0_0, all(X == [cell(0,1)])) :- connected(X, cell(0,0)).
+
+% Find all cells connected to/from 1,3
+test(connected_from_1_2, set(X == [cell(0,2), cell(1,1), cell(2,2)])) :-
+  connected(cell(1,2), X).
+test(connected_to_1_2, set(X == [cell(0,2), cell(1,1), cell(2,2)])) :-
+  connected(X, cell(1,2)).
+
+:- end_tests(connected).
+
+
+% Step 2: path_to/3
+%
+% Now create path_to/3 which takes three arguments:
+%  1. A start cell
+%  2. An end cell
+%  3. A path from the start cell to the end cell as a list.
+%
+%  Tips:
+%    1. You will probably need to use an accumulator
+%    2. It may be easier to construct the path in reverse and then reverse it
+%    3. You can use negation to eliminate paths that backtrack on themselves
+%
+% Ensure that the tests pass with no warnings.
+%
+
+% (TODO): find what is wrong with path to self.
+%path_to(From,To,Path) :- connected(From,To); connected(To,From);
+%                         connected(To,Path);
+%			 connected(From,Path).
+
+
+%path_to(X,Y,[X,Y]):-corridor(X,Y),!.
+%path_to(X,Y,[X|R]):-corridor(X,Z),path_to(Z,Y,R),!.
+%
+
+path_to(From,To,Path) :-
+	travel(From,To,[From],Q),
+	rev(Q,Path),!.
+
+travel(From,To,Path,[To|Path]) :-
+	connected(From,To),!.
+
+
+travel(From,To,Visited,Path) :-
+	connected(From,C),
+	C\==To,
+	not(mem(C,Visited)),
+	travel(C,To,[C|Visited],Path),!.
+
+
+:- begin_tests(path_to).
+
+% verify that a path from a cell to itself consists of just the one cell
+test(path_to_self) :-
+  path_to(cell(0,0), cell(0,0), [cell(0,0)]).
```

Comment:
The reason why this test was unable to pass is because you start your accumulator with a non-empty path that you try to append to.